server/src/modules/assets/assets.service.ts

```diff
@@ -220,4 +222,51 @@ export default class AssetService {
 			}
 		});
 	}
+
+	public static copy(url: string) {
+		return new Promise<string>(async (resolve, reject) => {
+			const bucket = process.env.ASSET_SERVER_BUCKET_NAME as string;
+			try {
+				const s3Client: S3 = await this.getS3Client();
+				const key: string | undefined = url.split(`/${bucket}/`).pop();
+				if (!key) {
+					throw new CustomError(
+						'Failed to copy file at url, because we failed to extract the file path from the url after the bucket location',
+						null,
+						{ bucketName: bucket, url }
+					);
+				}
+				const newId = getUuid();
+				const parts = key.split('.');
+				const newKey = parts[0].substring(0, parts[0].length - 35) + newId + '.' + parts[1];
+
+				s3Client.copyObject(
+					{
+						Key: newKey,
+						Bucket: bucket,
+						CopySource: `${bucket}/${key}`,
+					},
+					(err: AWSError, data) => {
+						if (err) {
+							const error = new ExternalServerError(
+								'Failed to copy asset from the s3 asset service',
+								err,
+								{
+									url,
+								}
+							);
```

Comment:
it might be handy to also see the params that were provided to the copyObject function
```suggestion
							const error = new ExternalServerError(
								'Failed to copy asset from the s3 asset service',
								err,
								{
									url,
									newKey,
									bucket,
									copySource: `${bucket}/${key}`
								}
							);
```